backend/service/topology/topology.go

```diff
@@ -79,6 +79,11 @@ func New(cfg *any.Any, logger *zap.Logger, scope tally.Scope) (service.Service,
 		return c, nil
 	}
 
+	// If TTL is not set default to two hours
+	if c.config.Cache.Ttl == nil {
+		c.config.Cache.Ttl = ptypes.DurationProto(time.Hour * 2)
+	}
```

Comment:
instead of storing/modifying the config, please add a `ttl time.Duration` member to the struct.